app/src/main/java/com/android/solvit/provider/ui/request/RequestsFeed.kt

```diff
@@ -54,40 +54,166 @@ import androidx.compose.ui.Alignment
-import androidx.compose.ui.window.Dialog
+import androidx.compose.ui.window.DialogProperties
 import androidx.lifecycle.viewmodel.compose.viewModel
 import coil.compose.AsyncImage
 import com.android.solvit.R
+import com.android.solvit.seeker.model.profile.SeekerProfile
+import com.android.solvit.seeker.model.profile.SeekerProfileViewModel
 import com.android.solvit.seeker.ui.navigation.BottomNavigationMenu
 import com.android.solvit.seeker.ui.provider.PackageCard
 import com.android.solvit.shared.model.NotificationsViewModel
 import com.android.solvit.shared.model.authentication.AuthViewModel
-import com.android.solvit.shared.model.map.Location
+import com.android.solvit.shared.model.chat.ChatViewModel
 import com.android.solvit.shared.model.packages.PackageProposal
 import com.android.solvit.shared.model.packages.PackageProposalViewModel
 import com.android.solvit.shared.model.request.ServiceRequest
 import com.android.solvit.shared.model.request.ServiceRequestStatus
 import com.android.solvit.shared.model.request.ServiceRequestViewModel
 import com.android.solvit.shared.model.service.Services
-import com.android.solvit.shared.ui.map.GetDirectionsBubble
 import com.android.solvit.shared.ui.navigation.LIST_TOP_LEVEL_DESTINATION_PROVIDER
 import com.android.solvit.shared.ui.navigation.NavigationActions
 import com.android.solvit.shared.ui.navigation.Screen
-import com.android.solvit.shared.ui.theme.Orange
 import java.text.SimpleDateFormat
 import java.util.Locale
 
-/** Composable function that displays the top bar of the requests feed screen. */
+/**
+ * Composable function that displays the list requests feed screen.
+ *
+ * @param serviceRequestViewModel The service request view model
+ * @param packageProposalViewModel The package proposal view model
+ * @param navigationActions The navigation actions
+ * @param notificationViewModel The notification view model
+ * @param authViewModel The authentication view model
+ * @param chatViewModel The chat view model
+ * @param seekerProfileViewModel The seeker profile view model
+ */
+@Composable
+fun RequestsFeedScreen(
+    serviceRequestViewModel: ServiceRequestViewModel =
+        viewModel(factory = ServiceRequestViewModel.Factory),
+    packageProposalViewModel: PackageProposalViewModel =
+        viewModel(factory = PackageProposalViewModel.Factory),
+    navigationActions: NavigationActions,
+    notificationViewModel: NotificationsViewModel =
+        viewModel<NotificationsViewModel>(factory = NotificationsViewModel.Factory),
+    authViewModel: AuthViewModel = viewModel<AuthViewModel>(factory = AuthViewModel.Factory),
+    chatViewModel: ChatViewModel = viewModel(factory = ChatViewModel.Factory),
+    seekerProfileViewModel: SeekerProfileViewModel =
+        viewModel(factory = SeekerProfileViewModel.Factory),
+) {
+  val allRequests by serviceRequestViewModel.requests.collectAsState()
+  val requests = allRequests.filter { it.status == ServiceRequestStatus.PENDING }
```

Comment:
Tell me if I am wrong, but couldn't we use the `pendingRequests` state from the `ServiceRequestViewModel` here?

Reply:
You are right!